cmd/buildah/common.go

```diff
@@ -206,63 +207,152 @@ func defaultFormat() string {
 // imageIsParent goes through the layers in the store and checks if i.TopLayer is
 // the parent of any other layer in store. Double check that image with that
 // layer exists as well.
-func imageIsParent(store storage.Store, topLayer string) (bool, error) {
-	children, err := getChildren(store, topLayer)
+func imageIsParent(ctx context.Context, sc *types.SystemContext, store storage.Store, image *storage.Image) (bool, error) {
+	children, err := getChildren(ctx, sc, store, image, 1)
 	if err != nil {
 		return false, err
 	}
 	return len(children) > 0, nil
 }
 
-// getParent returns the image ID of the parent. Return nil if a parent is not found.
-func getParent(store storage.Store, topLayer string) (*storage.Image, error) {
+func getImageConfig(ctx context.Context, sc *types.SystemContext, store storage.Store, imageID string) (*imgspecv1.Image, error) {
+	ref, err := is.Transport.ParseStoreReference(store, imageID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to parse reference to image %q", imageID)
+	}
+	image, err := ref.NewImage(ctx, sc)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to open image %q", imageID)
+	}
+	config, err := image.OCIConfig(ctx)
+	defer image.Close()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to read configuration from image %q", imageID)
+	}
+	return config, nil
+}
+
+func historiesDiffer(a, b []imgspecv1.History) bool {
+	if len(a) != len(b) {
+		return true
+	}
+	i := 0
+	for i < len(a) {
+		if a[i].Created == nil && b[i].Created != nil {
+			break
+		}
+		if a[i].Created != nil && b[i].Created == nil {
+			break
+		}
+		if a[i].Created != nil && b[i].Created != nil && !a[i].Created.Equal(*(b[i].Created)) {
+			break
+		}
+		if a[i].CreatedBy != b[i].CreatedBy {
```

Comment:
I think you could drop the first two checks and add the below here.
```
	if a[i].Created == nil || b[i].Created == nil {
			break
	}
```
But I do have "it's almost dinner time brain".

Reply:
No I think the code is correct.  But I believe if both are nil then he wants to continue checking.

Reply:
oh yeah, if both are nil then they fall through and probably should.  Ignore me.